### PR TITLE
Run oracle from inside cockatrice and improve sets handling; 

### LINKED ISSUE #135: launch oracle from cockatrice
instead of telling a user to launch oracle, launch it for them

### LINKED ISSUE #641: Add an additional option under the "Help" dropdown in-client to inform users on how to run Oracle
With this, it would cut down on how many people ask "how to update to (insert set here)" questions. If not that, then at least make explaining it easier (just click "Help" at the top).

### LINKED ISSUE #1072: Disabling a set in the "order sets" window has no effect
Unchecking a set's checkbox in the order sets window doesn't do anything - the images from that set will still be displayed. To truly disable images from a certain set, you have to put it at the bottom of the order sets window.

### PR DESCRIPTION
**Pull request by ctrlaltca**
This is a continuation of #891.
It adds the ability to run oracle from inside cockatrice and update the card database. Cockatrice will detect when oracle has finished its job, reload the card database and check if new sets have been found.
I also added a quick informative popup for users running cockatrice for the first time/updating from an older version, hoping it will be helpful; improvements to the message are welcome.
This possibly fixes parts of #135, #210, #641

Fix #135 
Fix #641 
Fix #1072

### CODE REVIEW

**Comment by ZeldaZach** [comment]
This is awesome! Great job, especially thinking about the new players

**Comment by tooomm** [comment]
Add information to the first-use-message that enabling/disabling doesn't change how images are displayed?
And that the order is important to affect which images are displayed.

Alternatively add this information to the `Edit sets...` dialog window above the list of sets.
I like this more I guess! The Edit sets windows shows up for the first run anyway. People might not remember about special handling of pictures etc. when they need it somehow later.

**Comment by ZeldaZach** [comment]
Can you give me steps to reproduce like I'm a new user? My attempts failed

**Comment by ctrlaltca** [comment]
Please see https://github.com/Cockatrice/Cockatrice/pull/891#issuecomment-91499679

**Comment by ZeldaZach** [comment]
I'm having issues testing this, so if somebody else would like to give it a shot that would rock!

File: cockatrice/src/tab_deck_editor.cpp [hunk 1]
```diff
@@ -786,39 +785,11 @@ void TabDeckEditor::filterRemove(QAction *action) {
...
-    // no set is enabled. Probably this is the first time running trice
-    if(!sets.getEnabledSetsNum())
+    if(db->hasDetectedFirstRun())
     {
-        sets.guessSortKeys();
-        sets.sortByKey();
-        sets.enableAll();
-        db->emitCardListChanged();
-
+        QMessageBox::information(this, tr("Welcome"), tr("Hi! Its seems like it's the first time you run this version of Cockatrice.<br/>All the sets in the card database have been enabled; if you want to hide a set from the deck editor, disable it in the \"Edit Sets\" window."));
```
Comment (tooomm) [inline cockatrice/src/tab_deck_editor.cpp]: \n over <br>?
At least that's what I've seen around in the client so far.
It gets recognized better by Transifex as well!

Will this message dialog shows up directly after the first app launch? Right before the `Edit sets...` pops up?

Not sure about the last part of the last sentence...
If you add detailed instructions/notes/information about how disabling+sorting sets effect the app in the `Edit sets...` dialog, that might be enough.
`"(...)\n All the sets in the card database have been enabled. Read more about changing set order or disabling specific sets in the the \"Edit Sets\" window."`

Comment (ctrlaltca) [inline cockatrice/src/tab_deck_editor.cpp]: > Use \n over <br>? It gets recognized better by Transifex as well!
Sure!

> Will this message dialog shows up directly after the first app launch? Right before the 
> Edit sets... pops up?
Exactly.

> "(...)\n All the sets in the card database have been enabled. Read more about changing the set order or disabling specific sets in the the \"Edit Sets\" window."

This one seems the nicer to me

**Comment by ctrlaltca** [comment]
Current window layout (thank you @tooomm for the mockup):
schermata 2015-05-14 alle 13 12 39

**Comment by tooomm** [comment]
Now addresses #1072 as well!

**Comment by tooomm** [comment]
error-card update

Cockatrice --> `Check card updates`
win 8, build 15dcfd1 (2015-05-20)

**Pull request merged**